[BUG]:  Invalid provider type: Provider source "opensearch-project/terraform-provider-opensearch"

### What is the bug?

Starting and using the provider locally in debug mode fails with this exception:

> Error parsing %!q(PANIC=String method: called String on zero-value addrs.Provider) as a provider address: Invalid provider type: Provider source "opensearch-project/terraform-provider-opensearch" has a type with the prefix "terraform-provider-", which isn't valid. Although that prefix is often used in the names of version control repositories for Terraform providers, provider source strings should not include it.
>
> Did you mean "opensearch-project/opensearch"?

The provider itself also prints an error message when started (`./terraform-provider-opensearch -debuggable`):

> {"@level":"info","@message":"2023/09/15 08:46:41 [ERROR] Error parsing provider name \"registry.terraform.io/opensearch-project/terraform-provider-opensearch\": Invalid provider type: Provider source \"opensearch-project/terraform-provider-opensearch\" has a type with the prefix \"terraform-provider-\", which isn't valid. Although that prefix is often used in the names of version control repositories for Terraform providers, provider source strings should not include it.\n\nDid you mean \"opensearch-project/opensearch\"?","@timestamp":"2023-09-15T08:46:41.397308+02:00"}

### How can one reproduce the bug?

Follow the steps described in the [Readme](https://github.com/opensearch-project/terraform-provider-opensearch#debugging-this-provider)

### What is the expected behavior?

Using the provider locally in debug mode works without errors.

### What is your host/environment?

MacOS (x86) 13.4.1, Terraform v1.4.5

--- a/main.go
+++ b/main.go
@@ -21,7 +21,7 @@ func main() {
 
 	if debugMode {
 		//nolint:staticcheck // SA1019 ignore this!
-		err := plugin.Debug(context.Background(), "registry.terraform.io/opensearch-project/terraform-provider-opensearch",
+		err := plugin.Debug(context.Background(), "registry.terraform.io/opensearch-project/opensearch",
 			&plugin.ServeOpts{
 				ProviderFunc: provider.Provider,
 			},

